projects/ui/src/state/beanstalk/sun/morning.ts

```diff
@@ -1,116 +1,122 @@
 
-export default function MorningUpdater() {
-  const morningTime = useSelector(selectMorningBlockTime);
-  const blockMap = useSelector(selectMorningBlockMap);
-  const morning = useSelector(selectMorning);
+  const [_, { calculate: calculateTemperature }] = useTemperature();
+  const [_soilData, { calculate: calculateNextSoil }] = useSoil();
 
-  const [fetchMorningField] = useFetchMorningField();
   const dispatch = useDispatch();
 
-  /// called when the state is notified that it needs to fetch for updates
-  /// If the block from on chain is greater than the block we have stored,
-  /// we know we have the most updated state.
-  const fetch = useCallback(async () => {
-    const [_adjustedTemp, _maxTemp, _soil, blockData] =
-      await fetchMorningField();
-    console.debug(`[MorningUpdater][fetch], blockData = `, blockData);
-
-    if (blockData?.blockNumber.gt(morning.blockNumber)) {
-      dispatch(setAwaitingMorningBlock(false));
-    }
-  }, [fetchMorningField, morning.blockNumber, dispatch]);
-
   useEffect(() => {
-    if (!morning.isMorning || !Object.keys(blockMap).length) return;
+    if (!morning.isMorning) return;
 
-    // set up the timer while in the morning state.
+    // set up the timer while in the  morning state.
     const intervalId = setInterval(async () => {
-      const _remaining = getDiffNow(morningTime.next);
-      const remainingSeconds = _remaining.as('seconds');
-      // If the lower limit hasn't been reached decrement the remaining time
-      if (remainingSeconds > 0) {
-        dispatch(setRemainingUntilBlockUpdate(_remaining));
-      }
-      // if the lower limit is reached, notify the state to
-      // fetch for updated block, temperature, & soil values
-      else if (remainingSeconds <= 0) {
-        dispatch(setAwaitingMorningBlock(true));
+      const { abovePeg, sunriseBlock, timestamp: sTimestamp } = season;
+      const { blockNumber: morningBlock } = morning;
+
+      const now = getNowRounded();
+      const _remaining = getDiffNow(morningTime.next, now);
+      if (
+        now.toSeconds() === morningTime.next.toSeconds() ||
+        _remaining.as('seconds') <= 0
+      ) {
+        const morningResult = getMorningResult({
+          timestamp: sTimestamp,
+          blockNumber: sunriseBlock,
+        });
+
+        const scaledTemp = calculateTemperature(morningBlock.plus(1));
+        const nextSoil = abovePeg ? calculateNextSoil(morningBlock) : undefined;
+
+        console.debug('[beanstalk/sun/useUpdateMorning]: new block: ', {
+          temp: scaledTemp.toNumber(),
+          soil: nextSoil?.toNumber() || 'N/A',
+          blockNumber: morningResult.morning.blockNumber.toNumber(),
+          index: morningResult.morning.index.toNumber(),
+          isMorning: morningResult.morning.isMorning,
+        });
+
+        dispatch(updateScaledTemperature(scaledTemp));
+        nextSoil && dispatch(updateTotalSoil(nextSoil));
```

Comment:
Should soil be split into a "total soil" and "scaled soil" like temperature is split?